### DISCUSSION TITLE
Progress and Resuming functionality for youtube content player

### DISCUSSION THREAD

**Discussion by pavankumar0408** [Sunbird-Knowlg/Community#161] [General Discussions]
Hi,

We have integrated YouTube content player in our Compass Application and now we are trying to get progress and implementing resume functionality for this. 

->Here, we are getting summary response object of the video progressed from telemetry end event but it has empty values in it (Screenshot attached below).

![image](https://github.com/Sunbird-Knowlg/Community/assets/131791648/395dd862-674e-460b-b51c-4d99c74812f1)

->Once if I get proper response from the end event I can update that information into content state update api and fetch details from content state read api. But here, if we have visitedLength info from read api, for resuming the video where user lefts previously now it should be played from there (Eg: video has 60 secs visitedLength 30 secs, in next visit player should play from 30th sec).I couldn't able figure it out the property that will resume the video. To achieve this functionality which property or data that I need to send from portal to player.

@vinukumar-vs

Thanks in Advance

**Comment by thillaibirla** [comment]
@Krishnaj20 - we are trying to enable resume feature for YouTube content, mp4 and scorm content.  We have some challenges resuming the content right now.  Would need help to resolve the same.  thanks.

**Comment by HarishGangula** [comment]
@pavankumar0408 To ensure that you receive the correct end event, kindly use the most recent version of the content player. We would like to inform you that, as of the current version, the content player does not support the resume functionality.
cc: @pallakartheekreddy

**Reply by Krishnaj20** [reply]
@pavankumar0408 @HarishGangula - I've asked Sunbird AI assistant the same query and I got the below response. 
<img width="1451" alt="Screenshot 2023-10-31 at 4 15 18 PM" src="https://github.com/Sunbird-Knowlg/Community/assets/62481664/234582cb-d246-449c-9da4-eb43d7bfa2fd">

Bot link is [here](http://20.244.48.128:8080/?uuid=a2b87882-6e6e-11ee-92a3-0242ac110002).

Although the resume functionality isn't currently available we will be more than happy to assist you in contributing this feature.

**Reply by pallakartheekreddy** [reply]
@pavankumar0408 Hope your query is now resolved. Please mark the thread as answered.

**Reply by pavankumar0408** [reply]
@Krishnaj20,,@HarishGangula  Thanks for letting know me about availability of feature. 
@HarishGangula , I am trying to update library but I am getting some error. I am not exactly sure that I am doing in right way. Could you please provide me steps / document for updating the library in Portal / links for setting up content player in locally. That helps me to check the progress of the video. 

Thanks in Advance
cc: @thillaibirla @pallakartheekreddy

**Reply by HarishGangula** [reply]
@pavankumar0408  Please check [here](https://www.npmjs.com/package/@project-sunbird/content-player/v/5.1.0) for the details on same.

**Reply by pallakartheekreddy** [reply]
@pavankumar0408 As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
